blazar/plugins/oshosts/host_plugin.py

```diff
@@ -350,6 +352,35 @@ def before_end(self, resource_id):
                 for server in client.servers.list(
                         search_opts={"node": host, "all_tenants": 1}):
                         client.servers.create_image(server=server)
+        if action == 'email':
+            reservation_id = host_reservation['reservation_id']
+            reservation = db_api.reservation_get(reservation_id)
+            lease = db_api.lease_get(reservation['lease_id'])
+            project_id = lease['project_id']
+            user_id = lease['user_id']
+            keystoneclient = keystone.BlazarKeystoneClient(username=CONF.os_admin_username,
+                                               password=CONF.os_admin_password,
+                                               tenant_name=CONF.os_admin_project_name)
+            project = keystoneclient.projects.get(project_id)
+            user = keystoneclient.users.get(user_id)
+            params_tmp = ('--to {recipient} '
+                          '--username {username} '
+                          '--project-name {project_name} '
+                          '--lease-name {lease_name} '
```

Comment:
This line is feeding user-generated data on the command line. I think there is a potential for them to hijack the script call to run some other code, e.g. if their lease is called `; touch /var/lib/blazar/foobar` what happens?

Please read the warning at https://docs.python.org/2/library/subprocess.html#frequently-used-arguments

Reply:
Great catch @priteau - I missed that. This is definitely susceptible to shell injection.